### PR TITLE
Update process user input

### PR DESCRIPTION
**Pull request by friday963**
# Important

**Please provide the requested parameters below to help us review your pull request.**

**Pull requests without a body or explanation will be rejected.**

Refactor updateuserinput func to consolidate logic and args setttings into smaller units of work.

## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added tests.
- [x] I have run `make check` and there are no failures.

## Type of change

Please delete options that are not relevant.

N/A

### CODE REVIEW

**Comment by friday963** [comment]
Figured I'd just start the PR so you could see the diff.

**Review by pouriyajamshidi — CHANGES_REQUESTED** [review]
Hey @friday963,

Thanks for your contribution. It definitely looks much cleaner than before.

After my requested changes are in place, this can get merged in.

File: tcping.go [hunk 1]
```diff
@@ -215,124 +215,146 @@ func usage() {
...
-	if *useIPv4 && *useIPv6 {
-		tcpStats.printer.printError("Only one IP version can be specified")
+func checkSetIPFlags(tcpstats *stats, ip4, ip6 *bool) {
+	// check if ip4 or ip6 are true, if so printError and exit
+	if *ip4 && *ip6 {
+		tcpstats.printer.printError("Only one IP version can be specified")
 		usage()
+	} else if *ip4 {
+		tcpstats.userInput.useIPv4 = true
+	} else if *ip6 {
```
[the remark this answers is not in the document]
Comment (friday963) [reply tcping.go:256]: Ok so split this into two methods.  One to check for both flags being true then print the error/usage if so, another to update the struct as a subsequent method right below?

func alertIPFlagConflict(){
if ip4 and 6
     printerror()
     useage()
}

func setIPFlags(){
if ip4
 useip4 = true
 if ip6
 useip6 = true
}

File: tcping.go [hunk 2]
```diff
@@ -215,124 +215,146 @@ func usage() {
...
-	if *useIPv4 {
-		tcpStats.userInput.useIPv4 = true
-	}
-
-	if *useIPv6 {
-		tcpStats.userInput.useIPv6 = true
-	}
+}
 
+func checkUpdatePort(tcpstats *stats, args []string) {
```
[the remark this answers is not in the document]
Comment (friday963) [reply tcping.go:262]: Good catch

File: tcping.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pouriyajamshidi) [reply tcping.go:256]: ah no, I meant to say those don't need to be `else if` just an `if` will do the trick.

**Review by pouriyajamshidi — APPROVED** [review]
Thanks @friday963!

**Comment by friday963** [comment]
My pleasure!

**Pull request merged**